connectors/sources/sharepoint_online.py

```diff
@@ -2457,4 +2481,18 @@ def _access_control_for_user(self, user):
         if email:
             user_access_control.append(_prefix_email(email))
 
+        if user_id:
+            user_access_control.append(_prefix_user_id(user_id))
+
         return user_access_control
+
+    async def _access_control_for_group_id(self, group_id):
+        if group_id.endswith("_o"):
+            # this isn't the group id, it's a reference to the _owners_ of this ID
+            real_group_id = group_id[0:-2]
+            access_control = []
+            async for owner in self.client.group_owners(real_group_id):
+                access_control.extend(await self._access_control_for_member(owner))
```

Comment:
Just raising this for visibility: `_access_control_for_group_id` can be invoked by `_access_control_for_member` which in turns can call `_access_control_for_group_id` - in certain edge case it can lead to infinite recursion, if: 
- Group A is owned by Group B.
- Group B is owned by Group A.

This doesn't sound possible but .. it's SPO, so would it be worth adding a check to completely eliminate the possibility of infinite recursion?

Reply:
> This doesn't sound possible but .. it's SPO

This is good justification. :)

Reply:
I got a solution all coded up to prevent infinite recursion, and then while writing a test, realized that it wasn't necessary, because the `/groups/<group_id>/owners` API [can only return Users](https://learn.microsoft.com/en-us/graph/api/group-list-owners?view=graph-rest-1.0&tabs=http#response), and not the generic "members" (which can be groups OR users). This let me swap the call to `_access_control_for_member` to `_access_control_for_user` which breaks the potential cycle of infinite recursion. 👍